### ISSUE TITLE
Exploring a base32 cidv1 hash shows links as cidv0

### ISSUE THREAD

**Issue by andrew** [ipfs-shipyard/ipfs-webui#1057]
**Describe the bug**

If you paste a cidv1 hash (`bafybeig2wjljwgppoadpk32x6iruxza6jdfmscehfzpdydg3wvhjju4woe` for example) into Explore, it shows `base32 - cidv1 - dag-pb - sha2-256-256` on the right but then when listing the linked data below it shows base58 cidv0 links (`QmadqAJ9rDUD7zdoyNc12gH4npVGLZN9V6fkybshEHanZz`). 

My CIDv1 was generated by adding a folder recursively with the following command in go-ipfs:

`ipfs add --cid-base=base32 --quiet -r tmp/1234/ipfs-npm-republish`

which shows a list of base32 cidv1 links in the output:

```
$ ipfs add --cid-base=base32 --quiet -r .
bafybeiaafcmlzszejclc6or5bptsrjfa4hageep2ifzvkrjfsvmgztrx3m
bafybeih7bmxy47lqljqtvfdguvetirmofiudj377e26xxl3xbf4ikqdye4
bafybeibheccf4lpvwtjp4dougkccacglje6srx7ks4wk5mbv2rg46srvfy
bafybeic2rs36j74kc3wo2tl6thqijhtm6xe6fvdxip3dlzvka6nwuqizv4
bafybeigdauopsh3d66lwdbuyshc6yvjt3i6pfd5huhcyuelvae2n6asgli
bafybeia4sbcaf6l2tn6c4t4rdpx3iahw2b6au3gc6jbgscywmklquudgoa
bafybeig4ebmsddkayym35j2xozbbdvv7n2tgfv5fgqxhksvapxuqggcike
bafybeidoe5z3eodw5eiiqxrj4bw2nh62z5iebx7625yl4orx2m3jlod2we
bafybeigxjhzm74vcugvpau57l6gyobejrzcbmt7nq662g3uhu4tcfz74j4
```

Which lead me to believe it's a bug in webui displaying it rather than in go-ipfs when adding the file.

**To Reproduce**
1. View a cidv1 hash (`bafyb...`) of a folder added to IPFS in webUI

**Expected behavior**

Show show cidv1s for all linked CIDs if the root was passed in as a cidv1

**Screenshots**

![Screenshot 2019-06-07 at 10 49 25](https://user-images.githubusercontent.com/1060/59096065-ee7b0600-8911-11e9-8245-5d497a4abcaf.png)

**Desktop (please complete the following information):**
 - OS: Mac OS
 - Browser: Firefox and IPFS Desktop
 - webui version: 2.4.6

**Comment by lidel** [comment]
@andrew IIUC this is not a bug. Those links are CIDv0s, and Web UI is showing it correctly.

Explainer: AFAIK `--cid-base=base32` is just a read-only porcelain to print all CIDs in base32 for user convenience. If CIDv0 is found, it is "updated" to CIDv1 before printing to the screen. You used it in `ipfs add`, but it does not make `add` use CIDv1 for internal representation when building DAG, it just converts all CIDs to CIDv1b32 before printing. 

Real root and internal links remain CIDv0, not CIDv1.
Compare the output without `--quiet`:

```console
$ echo "lol" | ipfs add --cid-base=base32
added bafybeif7ztnhq65lumvvtr4ekcwd2ifwgm3awq4zfr3srh462rwyinlb4y QmbFMke1KXqnYyBBWxB74N4c5SBnJMVAiMNRcGu6x1AwQH

$ echo "lol" | ipfs add --cid-version 1
added bafkreihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku bafkreihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku
```

**tl;dr** to opt-in for CIDv1 during `ipfs add` you need to use `--cid-version 1`:

```console
$ ipfs add --cid-version 1 --quiet -r tmp/1234/ipfs-npm-republish
```

**Comment by andrew** [comment]
That's mighty confusing UX from the CLI but not a bug in webUI 👍 

![Screenshot 2019-06-07 at 14 00 32](https://user-images.githubusercontent.com/1060/59105787-a2898a80-892c-11e9-9375-d3dc0332acc4.png)
